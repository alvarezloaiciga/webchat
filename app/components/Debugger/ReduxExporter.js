// @flow

import React from 'react';
import {getFormattedDateAndTime} from 'utils/utils';
import {saveAs} from 'file-saver';

type ReduxExporterProps = {
  monitorState?: {initialScrollTop: number}, // eslint-disable-line react/no-unused-prop-types
  actionsById?: Object,
  nextActionId?: number,
  stagedActionIds?: Array<number>,
  skippedActionIds?: Array<number>,
  committedState?: Object,
  computedStates?: Array<Object>, // eslint-disable-line react/no-unused-prop-types
  currentStateIndex?: number,
};

export class ReduxExporter extends React.Component {
  props: ReduxExporterProps;

  exportReduxState = () => {
    saveAs(
      new Blob([JSON.stringify(this.getStateAndActions(), null, 2)], {
        type: 'application/json;charset=utf-8',
      }),
      `chat-${getFormattedDateAndTime(new Date())}`,
    );
  };

  getStateAndActions = () => ({
    monitorState: this.props.monitorState,
    actionsById: this.props.actionsById,
    nextActionId: this.props.nextActionId,
    stagedActionIds: this.props.stagedActionIds,
    skippedActionIds: this.props.skippedActionIds,
    committedState: this.props.committedState,
    computedStates: [],
    currentStateIndex: this.props.currentStateIndex,
  });

  render() {
    return (
      <div className="ReduxExporter">
        <i
          className={`fa fa-download icon`}
          title="Export Redux State"
          onClick={this.exportReduxState}
        />
      </div>
    );
  }
}

export default ReduxExporter;
