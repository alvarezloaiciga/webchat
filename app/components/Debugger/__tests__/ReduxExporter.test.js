// @flow
import React from 'react';
import ReduxExporter from '../ReduxExporter';
import {shallow} from 'enzyme';
import type {ShallowWrapper} from 'enzyme';

describe('ReduxExporter component', () => {
  let wrapper: ShallowWrapper;
  let render: () => void;

  beforeEach(() => {
    render = () => {
      wrapper = shallow(<ReduxExporter />);
    };
  });

  describe('rendering', () => {
    beforeEach(() => {
      render();
    });

    it('renders with default props', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
