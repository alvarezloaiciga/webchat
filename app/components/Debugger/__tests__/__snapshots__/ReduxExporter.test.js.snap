// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxExporter component rendering renders with default props 1`] = `
<div
  className="ReduxExporter"
>
  <i
    className="fa fa-download icon"
    onClick={[Function]}
    title="Export Redux State"
  />
</div>
`;
